File: packages/rune-filesystem/android/RuneFileSystem/src/main/java/dev/rune/filesystem/RuneFileSystem.kt
package dev.rune.filesystem

import android.app.Activity
import android.util.Log
import com.rune.kit.runtime.RuneRuntime

/**
 * Public interface for RuneFileSystem module.
 */
object RuneFileSystem {
    private var moduleInstance: RuneFileSystemModule? = null

    /**
     * Initialize the FileSystem module with an Activity and RuneRuntime instance.
     */
    @JvmStatic
    fun initialize(activity: Activity, runtime: RuneRuntime) {
        if (moduleInstance != null) {
            Log.w("RuneFileSystem", "Module already initialized")
            return
        }

        moduleInstance = RuneFileSystemModule(activity.applicationContext)
        runtime.installModules(listOf(moduleInstance!!))
        Log.d("RuneFileSystem", "Module initialized")
    }

    /**
     * Clean up the module (called during app teardown).
     */
    @JvmStatic
    fun cleanup() {
        moduleInstance = null
    }
}

File: packages/rune-filesystem/ios/RuneFileSystem.swift
//
//  RuneFileSystem.swift
//  RuneFileSystem
//
//  Public interface for RuneFileSystem module
//

import Foundation
import RuneKit

@objc(RuneFileSystem)
public class RuneFileSystem: NSObject {
  private static var moduleInstance: RuneFileSystemModule?

  @objc public static func initialize(with runtime: RuneRuntime) {
    guard moduleInstance == nil else {
      print("[RuneFileSystem] Module already initialized")
      return
    }

    let module = RuneFileSystemModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[RuneFileSystem] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}
